integrationtests/testground/main.go

```diff
@@ -105,40 +110,56 @@ func testSyncThreads(env *runtime.RunEnv, ic *run.InitContext) (err error) {
 		env.RecordFailure(fmt.Errorf(msg, args...))
 	}
 	setup(env, ic)
-	for i, pair := range simulatedNetworks {
-		round := pair.simulation
-		timeout, err := time.ParseDuration(env.StringParam("test-timeout"))
-		if err != nil {
-			timeout = time.Minute
-		}
-		ctx, cancel := context.WithTimeout(context.Background(), timeout)
-		if ic.GlobalSeq == 1 {
-			msg("################### Round %s network ###################", round)
-			ic.NetClient.MustConfigureNetwork(ctx, &network.Config{
-				Network:        "default",
-				Enable:         true,
-				Default:        pair.shape,
-				CallbackState:  sync.State("network-configured-" + round),
-				CallbackTarget: 1,
-				RoutingPolicy:  network.AllowAll,
-			})
-			msg("Done configuring %s network", round)
-		}
+	successState := sync.State("testSyncThreads-success")
+	failState := sync.State("testSyncThreads-fail")
+	barrierSuccess := ic.SyncClient.MustBarrier(context.Background(), successState, env.TestInstanceCount)
+	barrierFail := ic.SyncClient.MustBarrier(context.Background(), failState, 1)
+	go func() {
```

Comment:
code is moved into goroutine unchanged. The purpose of this change is to either wait for all instances to succeed, so the test results are printed at the same place, or exit the whole test when any of the instance fails.